SMLHelper/Patchers/SpritePatcher.cs

```diff
@@ -14,19 +16,45 @@
 
     internal class SpritePatcher
     {
-        internal static void Patch()
+
+        internal static void PatchCheck(SpriteManager.Group group, string name)
+        {
+            if (string.IsNullOrEmpty(name) || !ModSprite.ModSprites.ContainsKey(group) || !ModSprite.ModSprites[group].TryGetValue(name, out _))
```

Comment:
Double key lookup pattern
> ContainsKey(group) + [group]

Replace with a TryGetValue for optimized usage.
```suggestion
            if (string.IsNullOrEmpty(name) || !ModSprite.ModSprites.TryGetValue(group, out var groupDict) || !groupDict.TryGetValue(name, out _))
```